### PR TITLE
refactored classes IcnsImageParser, PhotometricInterpreterLogLuv, ByteSourceArray, PcxWriter, pivotProcessor, ImageParser, PsdImageParser, PngImageParser for code smells. 

### PR DESCRIPTION
**Pull request by NaveenKunapaneni**
IcnsImageParser -> Refactored a complex method(writeImage) to multiple smaller components (getImageType(src), writeHeader(bos, imageType), writeImageData(bos, src, imageType), writeMaskData(bos, src, imageType),
 
PhotometricInterpreterLogLuv ->Introduced a new variable expected_bit_per_sample, 

ByteSourceArray -> Deomposed a complex contional if statement in getBlock method, 

PcxWriter -> Moved a method writePixel32 from PcxWriter to RleWriter class, 

pivotProcessor -> Extracted methods which calculates pivot related values and pushed them to a new class , 

ImageParser -> pushed down a method form the parent class (ImageParser) to child ( PsdImageParser PngImageParser) classes.

 I am currently studying on smells and various techniques to refactor smells. This is a small effort on getting an under standing code smells and refactoring them.

### CODE REVIEW

File: .idea/misc.xml
Comment (kinow) [inline .idea/misc.xml:1]: Ide files are not included with imaging.

File: src/main/java/org/apache/commons/imaging/color/ColorConversions.java [hunk 1]
```diff
@@ -49,9 +43,9 @@ public static ColorCieLab convertXYZtoCIELab(final double X, final double Y, fin
         double var_Z = Z / REF_Z; // REF_Z = 108.883
 
         // Pivot XÝZ:
-        var_X = pivotXYZ(var_X);
-        var_Y = pivotXYZ(var_Y);
-        var_Z = pivotXYZ(var_Z);
+        var_X = pivotProcessor.pivotXYZ(var_X);
+        var_Y = pivotProcessor.pivotXYZ(var_Y);
+        var_Z = pivotProcessor.pivotXYZ(var_Z);
```
Comment (kinow) [inline src/main/java/org/apache/commons/imaging/color/ColorConversions.java:48]: Variables in Imaging may have weird names but these many times match what is described on papers and specifications. Refactorings like these are not really helpful.

**Review by kinow — CHANGES_REQUESTED** [review]
Hi. Had a quick look but immediately noticed IDE files added to source and changes to code like remove unused variables, comments to with links to docs, and renaming variables.

These are not really helpful since the code in Imaging normally matches code in other implementations, or pseudocode from papers and spwcifications. Renaming x1 to something more meaningful is normally a good pracrice. But in the case of Imaging it is more important to quickly know what is the x1 mentioned in some spec or soc. We can still improve this with comments (these are always welcome).  Will have a look later to see if there are other changes that can be added, but these changes (ide and IDE/linter batch renaming and removal) would need to be dropped.

Thanks!!!

**Comment by garydgregory** [comment]
I agree with @kinow

**Comment by garydgregory** [comment]
Closing: No reply from the author.

**Pull request closed without merging**